shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/product_details/ProductDetailsViewModel.kt

```diff
@@ -45,57 +61,151 @@ class ProductDetailsViewModel(
                 ProductDetailsState.Event.NavigateToConsumerCart
             }
 
+            is ProductDetailsState.Action.AdditionClick -> selectAddition(
+                uuid = action.uuid,
+                groupUuid = action.groupUuid
+            )
+
             is ProductDetailsState.Action.Init -> {
                 observeCart()
-                getMenuProduct(menuProductUuid = action.menuProductUuid)
+                getMenuProduct(
+                    menuProductUuid = action.menuProductUuid,
+                    selectedAdditionUuidList = action.selectedAdditionUuidList
+                )
             }
         }
     }
 
-    fun getMenuProduct(
+    private fun selectAddition(uuid: String, groupUuid: String) {
+        setState {
+            val changedGroup = getChangedGroup(menuProduct, groupUuid = groupUuid, uuid = uuid)
+
+            val newAdditionGroups = menuProduct.additionGroups.mapNotNull { additionGroup ->
```

Comment:
1. Looks like business logic. Let's move to separate usecase and test it. 
2. Let's simplify it a little bit:
```
val newAdditionGroups = menuProduct.additionGroups.map { additionGroup ->
    if (additionGroup.uuid == groupUuid) {
        additionGroup.copy(
            additionList = if (additionGroup.singleChoice) {
                additionGroup.additionList.map { addition ->
                    addition.copy(isSelected = addition.uuid == uuid)
                }
            } else {
                additionGroup.additionList.map { addition ->
                    if (addition.uuid == uuid) {
                        addition.copy(isSelected = !addition.isSelected)
                    } else {
                        addition
                    }
                }
            }
        )
    } else {
        additionGroup
    }
}
```